### DISCUSSION TITLE
Using fleek across MacOS and Linux

### DISCUSSION THREAD

**Discussion by qnm** [ublue-os/fleek#308] [Q&A]
TL;DR Is there a way to manage packages across multiple platforms without using custom nix?

I'm using Fleek across multiple devices, mostly Linux but also a MacOS machine.

When I use `fleek add` for a particular package, if that package doesn't exist in both Linux and MacOS Nix repos, then my `fleek apply` will fail on the device without the package.

I'm not sure what the correct behaviour here should be. In home-manager I had a conditional statement, but I prefer the simplicity of Fleek as it covers 95% of my needs.

**Comment by naudster** [comment]
I'd also appreciate feedback on this problem. 

If we rely on including arch specific packages in custom.nix, what's the syntax? I'm pretty new to nix and can't figure it out. Thanks a lot.

**Reply by naudster** [reply]
Figured it out. Example custom.nix to include the `barrier` package:
```
{ pkgs, misc, ... }:
{
  # FEEL FREE TO EDIT: This file is NOT managed by fleek. 
  home.packages = with pkgs;  [
    barrier
  ];
}
```

As @qnm asks, a better solution would be great. Maybe allowing `packages` under `systems` in fleek.yml?

```
systems:
    - arch: x86_64
      hostname: alien
      os: linux
      packages:
          - barrier
```
